code/game/machinery/tcomms/tcomms_base.dm

```diff
@@ -296,9 +296,6 @@ GLOBAL_LIST_EMPTY(tcomms_machines)
 
 		for(var/obj/item/radio/R in new_connection.devices["[RADIO_CHAT]"])
 
-			if(istype(R, /obj/item/radio/headset))
```

Comment:
This feels like it might have some wider consequences. Just to be sure, have you tested that this won't cause weirdness with transmissions over intercoms reaching headsets they aren't supposed to, and such?

Reply:
If my understanding of Telecomms code is correct (Which may not be the case, because it is telecomms code) this list gotten by `new_connection.devices["[RADIO_CHAT]"]` is already filtered down to radios that share the radio frequency with the sender so it shouldn't affect that in any way. That just leaves any headsets that somehow get `tcm.data` set to `SIGNALTYPE_INTERCOM_SBR`, this is only done in when the radio has `requires_tcomms` set to FALSE.
https://github.com/ParadiseSS13/Paradise/blob/616f40b12d8522e196b5a457bc4e603643b8a253/code/game/objects/items/devices/radio/radio_objects.dm#L479-L484

I haven't noticed any unusual behaviour during testing and debugging, and as far as I can tell there shouldn't be any issues (Though you can never be sure.)

Reply:
I did some looking through as well, and I think you're right. Would like another set of eyes on it, but I think it's probably good.